libs/community/langchain_community/chains/pebblo_retrieval/base.py

```diff
@@ -405,11 +412,14 @@ def _send_prompt(self, qa_payload: Qa) -> None:
             logger.warning("An Exception caught in _send_discover: local %s", e)
 
         if self.api_key:
+            if not retrieval_data:
+                retrieval_data
             try:
                 headers.update({"x-api-key": self.api_key})
                 pebblo_cloud_url = f"{PEBBLO_CLOUD_URL}{PROMPT_URL}"
+                retrieval_data.update({"classified": qa_payload.classified, "name": self.app_name})
                 pebblo_cloud_response = requests.post(
-                    pebblo_cloud_url, headers=headers, json=qa_payload.dict(), timeout=20
+                    pebblo_cloud_url, headers=headers, json=retrieval_data, timeout=20
```

Comment:
1. return retrieval_data ?
2. if retrieval_data is None or anything other that json, api call might fail